cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -329,18 +329,30 @@ private long computeChecksum() {
       computeAndValidateFieldLineageInfo(this.operations);
     }
 
+    Map<String, Set<EndPointField>> operationEndPointMap = new HashMap<>();
     Map<EndPointField, Set<EndPointField>> summary = new HashMap<>();
     for (WriteOperation write : writeOperations) {
       List<InputField> inputs = write.getInputs();
       for (InputField input : inputs) {
-        computeIncomingSummaryHelper(new EndPointField(write.getDestination(), input.getName()),
-                                     operationsMap.get(input.getOrigin()), write, summary);
+        EndPointField dest = new EndPointField(write.getDestination(), input.getName());
+        if (operationEndPointMap.containsKey(input.getOrigin())) {
+          summary.put(dest, new HashSet<>(operationEndPointMap.get(input.getOrigin())));
+          continue;
```

Comment:
Better not to use continue in this case, but use an `else` clause for one liner if-else like in here.